spec/models/account_spec.rb

```diff
@@ -1,34 +1,87 @@
 describe Account do
-  before(:each) do
-    @user = FactoryGirl.create(:account_user)
-    @group = FactoryGirl.create(:account_group)
+  let(:user) { FactoryGirl.create(:account_user) }
+  let(:group) { FactoryGirl.create(:account_group) }
+
+  context 'when host has users and groups' do
+    let(:host) { FactoryGirl.create(:host) }
+    let(:test_xml) do
+      REXML::Document.new("<miq>
+                          <accounts>
+                          <users><user name='test_user'/></users>
+                          <groups><group name='test_group'/></groups>
+                          </accounts>
+                          </miq>")
+    end
+
+    it '.add_elements' do
+      user.class.add_elements(host, test_xml)
+      expect(host.users.find_by(:name => 'test_user')).to be
+      expect(host.groups.find_by(:name => 'test_group')).to be
+    end
   end
 
   it '#accttype_opposite' do
-    expect(@user.class.accttype_opposite('user')).to eq('group')
-    expect(@group.accttype_opposite).to eq('user')
+    expect(group.accttype_opposite).to eq('user')
+    expect(user.accttype_opposite).to eq('group')
   end
 
-  it '#users raise an error when called on user model' do
-    expect { @user.users }.to raise_error(RuntimeError, "Cannot call method 'users' on an Account of type 'user'")
+  it 'users raise an error when called on user model' do
+    expect { user.users }.to raise_error(RuntimeError, "Cannot call method 'users' on an Account of type 'user'")
   end
 
-  it '#users return empty array when called on group' do
-    expect(@group.users).to be_empty
+  it 'users return empty array when called on group' do
+    expect(group.users).to be_empty
   end
 
-  it '#groups raise an error when called on group model' do
-    expect { @group.groups }.to raise_error(RuntimeError, "Cannot call method 'groups' on an Account of type 'group'")
+  it 'groups raise an error when called on group model' do
+    expect { group.groups }.to raise_error(RuntimeError, "Cannot call method 'groups' on an Account of type 'group'")
   end
 
-  it '#groups returns empty array when called on user model' do
-    expect(@user.groups).to be_empty
+  it 'groups returns empty array when called on user model' do
+    expect(user.groups).to be_empty
+  end
+
+  describe 'group methods' do
+    let(:user1) { FactoryGirl.create(:account_user) }
+
+    it '#add_user' do
+      expect(group.add_member(user)).not_to be_empty
+      expect(group.add_member(user1)).not_to be_empty
+      # should not add new user if the user already exist in @group
+      expect(group.add_member(user)).to be_empty
+      expect(group.members).to include(user, user1)
+    end
+
+    it '#remove_user' do
+      expect(group.remove_member(user)).to be_empty
+      expect(group.members).not_to include(user)
```

Comment:
Is this a false positive? You didn't add `user` to the group in this example, so removing it from the group should be empty and it shouldn't include `user` anyway.